incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -34,15 +36,24 @@ var (
 var treeCmd = &cobra.Command{
 	Use:   "tree",
 	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		nnm := args[0]
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		hier := getHierarchy(nnm)
-		fmt.Println(nameAndFootnotes(hier))
-		printSubtree("", hier)
-
+		nsList := make([]string, 0)
+		if len(args) != 0 {
+			nsList = append(nsList, args[0])
+		} else {
+			allNamespaces := getAllNamespaces()
+			nsList = append(nsList, allNamespaces...)
+		}
+		for _, nnm := range nsList {
+			hier := getHierarchy(nnm)
+			if (hier.Spec.Parent == "" && len(args) == 0) || len(args) != 0 {
```

Comment:
In Go, it's usually considered better style to early exit. If you use the `defaultList` suggestion above, this then becomes:

```
if defaultList && hier.Spec.Parent != "" {
  // By default, skip non-root namespaces since they'll be displayed as part of
  // another namespace's tree.
  continue
}
```

which is a much easier flow of logic to follow, IMO.